src/calculator/index.ts

```diff
@@ -266,10 +277,138 @@ export default class Calculator {
     return augmented;
   }
 
+  /**
+   * Estimates matching for a given project and potential additional votes
+   * @param potentialVotes
+   * @param roundId
+   */
+  async estimateMatching(
+    potentialVotes: PotentialVote[],
+    roundId: string
+  ): Promise<MatchingEstimateResult[]> {
+    const votes = await this.parseJSONFile<Vote>(
+      "votes",
+      `${this.chainId}/rounds/${this.roundId}/votes.json`
+    );
+
+    const rounds = await this.parseJSONFile<Round>(
+      "rounds",
+      `${this.chainId}/rounds.json`
+    );
+
+    const round = rounds.find((r: Round) => r.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
+
+    if (round.matchAmount === undefined) {
+      throw new ResourceNotFoundError("round match amount");
+    }
+
+    if (round.token === undefined) {
+      throw new ResourceNotFoundError("round token");
+    }
+
+    const applications = await this.parseJSONFile<Application>(
+      "applications",
+      `${this.chainId}/rounds/${this.roundId}/applications.json`
+    );
```

Comment:
we can skip all these validations since they are already done in `_calculate`. 
It's better to have a single point with this logic, if we add more validations in `_calculate` we might forget to add them here as well.

Reply:
Fixed.